lib/syntax_tree/with_scope.rb

```diff
@@ -154,7 +154,13 @@ def visit_params(node)
       add_argument_definitions(node.requireds)
 
       node.posts.each do |param|
```

Comment:
We're going to have the same issues with splats that we had for requireds here.
```suggestion
        add_argument_definitions(node.posts)
```

Reply:
Can you share the test example that will trigger it? Is it

```ruby
def foo(a, *b)
end
```

Reply:
```ruby
def foo(optional = 1, (bin, bag, *))
end
```

because they can be nested, it could also be:

```ruby
def foo(optional = 1, (bin, (bag)))
end
```

So effectively they should be treated exactly as the other required parameters.